### ISSUE TITLE
Scatterplot fails with matplotlib==3.3.1

### ISSUE THREAD

**Issue by MaozGelbart** [mwaskom/seaborn#2194]
Matplotlib 3.3.1 has just been released and it breaks `scatterplot` (and `relplot(kind="scatter")`). The problem is the `marker=''` argument in the legend artists creation, which now raises an exception upon drawing (seems unintentional). See [https://github.com/matplotlib/matplotlib/issues/18254](https://github.com/matplotlib/matplotlib/issues/18254). Example code:

```python
import seaborn as sns

# Load the example mpg dataset
mpg = sns.load_dataset("mpg")

# Plot miles per gallon against horsepower with other semantics
sns.scatterplot(x="horsepower", y="mpg", hue="origin", data=mpg)
```

Until upstream issue is resolved I'd advocate using matplotlib==3.3.0.

**Comment by trenton3983** [comment]
Here is a work around for now

- Use `hue=mpg.origin.tolist()`

`sns.scatterplot(x="horsepower", y="mpg", hue=mpg.origin.tolist(), data=mpg)`

**Comment by mwaskom** [comment]
(The explanation for the workaround is that the artist that triggers the bug is added as part of the legend "title", entry, and using a list (or numpy array) won't put a title in the legend because the vector has no name).

I think I've actually inadvertently fixed this in #2229, actually.

**Comment by trenton3983** [comment]
Michael, that makes sense. I didn't pay attention to the fact that the legend doesn't have a title with the work around. Thanks for your continued hard work on Seaborn. It's a great package.

**Comment by MaozGelbart** [comment]
`matplotlib==3.3.2` has been released and I confirm the issue is resolved for `seaborn==0.10.1`.
